rust/src/ssh/ssh.rs

```diff
@@ -179,8 +194,19 @@ impl SSHState {
                             let remlen = rem.len() as u32;
                             hdr.record_left = head.pkt_len - 2 - remlen;
                             //header with rem as incomplete data
-                            if head.msg_code == SSH_MSG_NEWKEYS {
-                                hdr.flags = SSHConnectionState::SshStateFinished;
+                            match head.msg_code {
+                                parser::MessageCode::SshMsgNewKeys => {
+                                    hdr.flags = SSHConnectionState::SshStateFinished;
+                                }
+                                parser::MessageCode::SshMsgKexinit if unsafe {SSHHasshIsEnabled()} && !input.is_empty() => {
+                                    match parser::parse_packet_key_exchange(&input[SSH_RECORD_HEADER_LEN..]) {
```

Comment:
Does it make sense to parse the incomplete `SshMsgKexinit` record ?

Reply:
yes, you're right, incorrect hassh can be created.